cpp/src/arrow/compute/light_array_test.cc

```diff
@@ -18,6 +18,7 @@
 #include "arrow/compute/light_array.h"
 
 #include <gtest/gtest.h>
+#include <numeric>
```

Comment:
+1 to this change.  I had noticed it myself but hadn't gotten around to making a PR.